lib/utils/injectRefreshLoader.js

```diff
@@ -38,6 +38,13 @@ function injectRefreshLoader(moduleData, injectOptions) {
     // Check to prevent double injection
     !moduleData.loaders.find(({ loader }) => loader === resolvedLoader)
   ) {
+    // if matchResource is set it means another loader is
+    // dynamically generating a module so let's look into moduleData.request
+    // to give users the opportunity to exclude them from instrumentation
+    if (moduleData.matchResource && !match(moduleData.request)) {
+      return moduleData;
+    }
```

Comment:
I think we can collapse this into the big if statement above?

Reply:
IMO readability is easier by doing it outside since we only want to match against `moduleData.request` if `moduleData.matchResource` is set but maybe my brain is just not working well right now lol

Reply:
I guess another way to do this is to break down the if statement and do a bunch of early returns 🤔